File: src/main/java/com/placesync/job/entity/Job.java
package com.placesync.job.entity;

import com.placesync.company.entity.Company;
import com.placesync.recruiter.entity.RecruiterProfile;
import com.placesync.user.entity.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "jobs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", nullable = false, updatable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "recruiter_id", nullable = false, foreignKey = @ForeignKey(name = "fk_jobs_recruiter"))
    private RecruiterProfile recruiter;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "company_id", nullable = false, foreignKey = @ForeignKey(name = "fk_jobs_company"))
    private Company company;

    @Column(name = "title", nullable = false, length = 255)
    private String title;

    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "location_type", nullable = false, columnDefinition = "job_location_type")
    private JobLocationType locationType;

    @Enumerated(EnumType.STRING)
    @Column(name = "job_type", nullable = false, columnDefinition = "job_type")
    private JobType jobType;

    @Column(name = "location_city", length = 255)
    private String locationCity;

    @Column(name = "compensation", length = 255)
    private String compensation;

    @Column(name = "application_deadline", nullable = false)
    private OffsetDateTime applicationDeadline;

    @Column(name = "min_cgpa", precision = 3, scale = 2)
    private BigDecimal minCgpa;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, columnDefinition = "job_status")
    @Builder.Default
    private JobStatus status = JobStatus.PENDING_APPROVAL;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "approved_by", foreignKey = @ForeignKey(name = "fk_jobs_approved_by"))
    private User approvedBy;

    @Column(name = "approved_at")
    private OffsetDateTime approvedAt;

    @Column(name = "closed_at")
    private OffsetDateTime closedAt;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false, insertable = false, updatable = false)
    private OffsetDateTime updatedAt;

    @Column(name = "deleted_at")
    private OffsetDateTime deletedAt;
}

File: placesync-frontend/src/types/job.ts
export type JobStatus = 'PENDING_APPROVAL' | 'OPEN' | 'CLOSED' | 'EXPIRED';
export type JobLocationType = 'ONSITE' | 'REMOTE' | 'HYBRID';
export type JobType = 'FULL_TIME' | 'INTERNSHIP' | 'CONTRACT';

export interface JobSummary {
  id: string;
  title: string;
  companyId: string;
  companyName: string;
  locationType: JobLocationType;
  jobType: JobType;
  locationCity?: string;
  compensation?: string;
  applicationDeadline: string;
  minCgpa?: number;
  status: JobStatus;
  createdAt: string;
}

export interface Job {
  id: string;
  title: string;
  description: string;
  companyId: string;
  companyName: string;
  recruiterId: string;
  recruiterFirstName: string;
  recruiterLastName: string;
  locationType: JobLocationType;
  jobType: JobType;
  locationCity?: string;
  compensation?: string;
  applicationDeadline: string;
  minCgpa?: number;
  status: JobStatus;
  requiredSkills: string[];
  eligibleDepartments: string[];
  approvedAt?: string;
  closedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface JobFilters {
  keyword?: string;
  locationType?: string;
  jobType?: string;
}
